### ISSUE TITLE
2.x doesn't work with Weblogic

### ISSUE THREAD

**Issue by apixandru** [spring-projects/spring-boot#10974]
It looks like weblogic deployment no longer works with Spring Boot 2.x

Sample project here:
https://github.com/apixandru/case-study/tree/master/spring-hibernate-transactional

Stacktrace:
[mydomain.log](https://github.com/spring-projects/spring-boot/files/1460373/mydomain.log)

**Comment by wilkinsona** [comment]
Spring Boot 2.0 required Java 8. It would appear that your WebLogic Server, specifically the repackaged version of ASM that it is using, doesn’t support Java 8. If you are not in a position to upgrade WebLogic you’ll have to stick with Spring Boot 1.5.

**Comment by apixandru** [comment]
Hi @wilkinsona,

Thanks for the hint, we are indeed using WebLogic 12.1.3 which is pretty old.

Unfortunately, I can't upgrade weblogic, but I can 'patch' it with the standard patching mechanism. It was indeed the repackaged asm library that is preventing the deployment. After I 'patched' Weblogic 12.1.3 with Weblogic 12.2.1's asm, it almost worked.

```
set WEBLOGIC_PATCH_DIR=*a directory where i put the new weblogic asm libraries*
set PATCH_CLASSPATH==%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.commons.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.tree.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.util.jar

```

After some tinkering i found out that in order to use spring boot 2.x.x these additional packages had to be preferred into the weblogic.xml and after that everything works as expected.

```
<wls:package-name>javax.validation</wls:package-name>
<wls:package-name>org.hibernate.validator</wls:package-name>
<wls:package-name>javax.el</wls:package-name>
<wls:package-name>javax.persistence</wls:package-name>
```

**Comment by jorgheymans** [comment]
@apixandru do you happen to know of a way to patch this without updating weblogic's internal asm version ? Did you try to put a dependency on asm in web-inf/lib and add it to preferred pkg ?

**Comment by apixandru** [comment]
@jorgheymans I did not go beyond patching it and unfortunately I've moved on since so I don't have an environment where i can test that at my disposal but feel free to try and let us know if it worked. It sounds like it should.

**Comment by Clydeocean** [comment]
> Hi @wilkinsona,
> 
> Thanks for the hint, we are indeed using WebLogic 12.1.3 which is pretty old.
> 
> Unfortunately, I can't upgrade weblogic, but I can 'patch' it with the standard patching mechanism. It was indeed the repackaged asm library that is preventing the deployment. After I 'patched' Weblogic 12.1.3 with Weblogic 12.2.1's asm, it almost worked.
> 
> ```
> set WEBLOGIC_PATCH_DIR=*a directory where i put the new weblogic asm libraries*
> set PATCH_CLASSPATH==%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.commons.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.tree.jar;%WEBLOGIC_PATCH_DIR%\com.bea.core.repackaged.asm.util.jar
> ```
> After some tinkering i found out that in order to use spring boot 2.x.x these additional packages had to be preferred into the weblogic.xml and after that everything works as expected.
> 
> ```
> <wls:package-name>javax.validation</wls:package-name>
> <wls:package-name>org.hibernate.validator</wls:package-name>
> <wls:package-name>javax.el</wls:package-name>
> <wls:package-name>javax.persistence</wls:package-name>
> ```

Hi, @jorgheymans , can you post your specific startWeblogic.cmd configuration?

**Comment by jorgheymans** [comment]
a bit late sorry @Clydeocean 

```
<?xml version="1.0" encoding="UTF-8"?>
<wls:weblogic-web-app
  xmlns:wls="http://xmlns.oracle.com/weblogic/weblogic-web-app"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://xmlns.oracle.com/weblogic/weblogic-web-app
        http://xmlns.oracle.com/weblogic/weblogic-web-app/1.4/weblogic-web-app.xsd">
  ....
  <wls:container-descriptor>
    <wls:prefer-application-resources>
      <wls:resource-name>org/slf4j/impl/StaticLoggerBinder.class</wls:resource-name>
    </wls:prefer-application-resources>
    <wls:prefer-application-packages>
      <wls:package-name>org.slf4j</wls:package-name>
      <wls:package-name>org.joda</wls:package-name>
      <wls:package-name>com.google.common</wls:package-name>
      <wls:package-name>org.springframework</wls:package-name>
      <wls:package-name>javassist</wls:package-name>
      <wls:package-name>org.apache.commons</wls:package-name>
      <wls:package-name>javax.validation</wls:package-name>
      <wls:package-name>org.hibernate.validator</wls:package-name>
    </wls:prefer-application-packages>
  </wls:container-descriptor>
</wls:weblogic-web-app>
```
